go/diff/patch.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package diff
+
+import (
+	"bytes"
+
+	"github.com/attic-labs/noms/go/types"
+)
+
+// Patch is a list of difference objects that can be applied to a graph
+// using ApplyPatch(). Patch implements a sort order that is useful for
+// applying the patch in an efficient way.
+type Patch []Difference
+
+func (r Patch) Swap(i, j int) {
+	r[i], r[j] = r[j], r[i]
+}
+
+func (r Patch) Len() int {
+	return len(r)
+}
+
+func (r Patch) Less(i, j int) bool {
+	if r[i].Path.Equals(r[j].Path) {
+		if r[i].ChangeType == r[j].ChangeType {
+			return false
+		}
+		if r[i].ChangeType == types.DiffChangeRemoved {
```

Comment:
To make this easier to read, you could have a map from these constants to numeric values, then just compare those directly.